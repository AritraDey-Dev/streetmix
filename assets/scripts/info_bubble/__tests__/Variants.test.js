/* eslint-env jest */
import React from 'react'
import { Variants } from '../Variants'
import { shallow } from 'enzyme'
import { IntlProvider } from 'react-intl'

import { getSegmentInfo } from '../../segments/info'
import {
  INFO_BUBBLE_TYPE_SEGMENT,
  INFO_BUBBLE_TYPE_LEFT_BUILDING,
  INFO_BUBBLE_TYPE_RIGHT_BUILDING
} from '../constants'

jest.mock('../../streets/data_model', () => {})
jest.mock('../../segments/view')
jest.mock('../../segments/info')

describe('Variants', () => {
  const intlProvider = new IntlProvider({ locale: 'en' }, {})
  const { intl } = intlProvider.getChildContext()
  let segment
  beforeEach(() => {
    segment = { variants: ['direction', 'public-transit-asphalt'] } // coming from info.json
    getSegmentInfo.mockImplementation(() => segment)
  })
  it('renders', () => {
    const type = INFO_BUBBLE_TYPE_SEGMENT
    const streetSegment = { variantString: 'inbound|regular', segmentType: 'streetcar' }
    const wrapper = shallow(<Variants intl={intl} type={type} position={0} variant={streetSegment.variantString} segmentType={streetSegment.segmentType} />)
    expect(wrapper).toMatchSnapshot()
  })
})
