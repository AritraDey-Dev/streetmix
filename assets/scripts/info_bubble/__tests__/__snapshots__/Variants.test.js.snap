// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Variants INFO_BUBBLE_TYPE_LEFT_BUILDING renders left building correctly 1`] = `
<div
  className="variants"
>
  <button
    disabled={false}
    key="building.waterfront"
    onClick={[Function]}
    title="Waterfront"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-waterfront"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.grass"
    onClick={[Function]}
    title="Grass"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-grass"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.fence"
    onClick={[Function]}
    title="Empty lot"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-fence"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.parking-lot"
    onClick={[Function]}
    title="Parking lot"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-car"
      />
    </svg>
  </button>
  <button
    disabled={true}
    key="building.residential"
    onClick={[Function]}
    title="Home"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-residential"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.narrow"
    onClick={[Function]}
    title="Narrow building"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-building-thin"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.wide"
    onClick={[Function]}
    title="Wide building"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-building-wide"
      />
    </svg>
  </button>
</div>
`;

exports[`Variants INFO_BUBBLE_TYPE_RIGHT_BUILDING renders left building correctly 1`] = `
<div
  className="variants"
>
  <button
    disabled={false}
    key="building.waterfront"
    onClick={[Function]}
    title="Waterfront"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-waterfront"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.grass"
    onClick={[Function]}
    title="Grass"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-grass"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.fence"
    onClick={[Function]}
    title="Empty lot"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-fence"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.parking-lot"
    onClick={[Function]}
    title="Parking lot"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-car"
      />
    </svg>
  </button>
  <button
    disabled={true}
    key="building.residential"
    onClick={[Function]}
    title="Home"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-residential"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.narrow"
    onClick={[Function]}
    title="Narrow building"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-building-thin"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="building.wide"
    onClick={[Function]}
    title="Wide building"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-building-wide"
      />
    </svg>
  </button>
</div>
`;

exports[`Variants INFO_BUBBLE_TYPE_SEGMENT renders segment type correctly 1`] = `
<div
  className="variants"
>
  <button
    disabled={true}
    key="direction.inbound"
    onClick={[Function]}
    title="Inbound"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-direction-inbound"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="direction.outbound"
    onClick={[Function]}
    title="Outbound"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-direction-outbound"
      />
    </svg>
  </button>
  <hr
    key="public-transit-asphalt"
  />
  <button
    disabled={true}
    key="public-transit-asphalt.regular"
    onClick={[Function]}
    title="Asphalt"
  >
    <svg
      className="icon"
      style={
        Object {
          "fill": "#292a29",
        }
      }
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-asphalt"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="public-transit-asphalt.colored"
    onClick={[Function]}
    title="Red lane"
  >
    <svg
      className="icon"
      style={
        Object {
          "fill": "#9b1f22",
        }
      }
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-asphalt"
      />
    </svg>
  </button>
  <button
    disabled={false}
    key="public-transit-asphalt.grass"
    onClick={[Function]}
    title="Grass"
  >
    <svg
      className="icon"
      style={null}
      xmlns="http://www.w3.org/1999/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <use
        xlinkHref="#icon-grass"
      />
    </svg>
  </button>
</div>
`;

exports[`Variants does not render component 1`] = `""`;
